index.html

```diff
@@ -1565,83 +1612,129 @@
+            <a>NFC adapter</a> to <var>output</var>.
+          </li>
+          <li id="adapter-timer">
+            If <var>options</var>.timeout value is not <code>Infinity</code>,
+            start the timer <var>timer</var>
+            <a href="#adapter-timer">associated</a> with the current
+            <a>NFC adapter</a>, with the timeout value set to
+            <var>options</var>.timeout.
           </li>
           <li>
-            When an <a>NFC device</a> comes within communication range,
+            Wait until one of the following events happens:
             <ul>
               <li>
-                If <var>target</var> has the value <code>"tag"</code>,
-                then the sending MAY only happen if an <a>NFC tag</a> is
-                tapped, otherwise reject <var>promise</var> with
-                <code>"InvalidAccessError"</code> and abort these steps.
-              </li>
-              <li>
-                If <var>target</var> has the value <code>"peer"</code>,
-                then the sending MAY only happen if an <a>NFC peer</a> is
-                tapped, otherwise reject <var>promise</var> with
-                <code>"InvalidAccessError"</code> and abort these steps.
-              </li>
-              <li>
-                If <var>target</var> has the value <code>"any"</code>
-                (which is the default value), then the sending should happen if
-                either an <a>NFC peer</a> or an <a>NFC tag</a> is tapped within
-                the specified timeout.
+                The <var>timer</var> expires, reject <var>promise</var>
+                with <code>"TimeoutError"</code> and abort these steps.
               </li>
               <li>
-                If the connection is lost during the transfer, or an error is
-                signaled by the underlying platform, reject
-                <var>promise</var> with <code>"<a>NetworkError</a>"</code> and
-                terminate this algorithm.
+                If <var>timer</var> is active, and the
+                <a href="cancelPush"><code>cancelPush()</code></a>
+                method is called, reject <var>promise</var> with
+                <code>"<a>AbortError</a>"</code>, as described in the
+                <a href="#steps-cancelPush"><code>cancelPush()</code>
+                algorithm</a>.
               </li>
               <li>
-                If the transfer is successful, cancel <var>timer</var> and
-                resolve <var>promise</var>.
+                When an <a>NFC device</a> <var>device</var> comes within
+                communication range,
+                <ol>
+                  <li>
+                    If <var>suspended</var> variable
+                    <a href="#adapter-suspended">associated</a> with the current
+                    <a>NFC adapter</a> is <code>true</code>, then
+                     reject <var>promise</var> with
```

Comment:
This looks like an information leak: while the page is in the background, it learns the instant an NFC radio comes nearby. Instead, maybe just ignore any nearby devices until the page is focused again?